From stern@rowland.harvard.edu  Sat Sep  8 07:42:37 2007
From: Alan Stern <stern@rowland.harvard.edu>
Date: Tue, 4 Sep 2007 10:12:44 -0400 (EDT)
Subject: USB: prevent Genesys USB-IDE from autosuspending
To: Greg KH <greg@kroah.com>
Cc: USB development list <linux-usb-devel@lists.sourceforge.net>
Message-ID: <Pine.LNX.4.44L0.0709041011570.5120-100000@iolanthe.rowland.org>


This patch (as986) prevents the troublesome Genesys USB-IDE adapter
from autosuspending.  It may not be necessary for all such devices,
but the one in Bugzilla #8892 sometimes fails to resume.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/quirks.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -70,6 +70,10 @@ static const struct usb_device_id usb_qu
 	{ USB_DEVICE(0x05cc, 0x2267), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 	/* Ultima Electronics Corp.*/
 	{ USB_DEVICE(0x05d8, 0x4005), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+
+	/* Genesys USB-to-IDE */
+	{ USB_DEVICE(0x0503, 0x0702), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
+
 	/* Agfa Snapscan1212u */
 	{ USB_DEVICE(0x06bd, 0x2061), .driver_info = USB_QUIRK_NO_AUTOSUSPEND },
 	/* Seagate RSS LLC */
